scripts/clustering.py

```diff
@@ -4,42 +4,62 @@
 """
 
 import argparse
+import os
 import sys
 
 import matplotlib.pyplot as plt
 import numpy as np
 
+from asaplib.data import ASAPXYZ
 from asaplib.pca import KernelPCA
 from asaplib.kde import KDE
-from asaplib.kernel import kerneltodis
 from asaplib.cluster import DBCluster, sklearn_DB, LAIO_DB
-from asaplib.plot import plot_styles
+from asaplib.plot import *
 from asaplib.io import str2bool
 
 
-def main(fmat, kmat, ftags, prefix, fcolor, dimension, pc1, pc2, algorithm, adtext):
+def main(fmat, fxyz, kmat, ftags, prefix, fcolor, colorscol, peratom, dimension, pc1, pc2, algorithm, projectatomic, plotatomic,
+         adtext):
 
     """
 
     Parameters
     ----------
-    fmat
-    kmat
-    ftags
-    prefix
-    fcolor
-    dimension
-    pc1
-    pc2
-    algorithm
-    adtext
+    fmat: Location of the low D projection of the data
+    fxyz: Location of xyz file for reading the properties.
+    kmat: Location of kernel matrix file. You can use gen_kmat.py to compute it
+    ftags: Location of tags for the first M samples
+    prefix: Filename prefix. Default is ASAP.
+    fcolor: Properties for all samples (N floats) used to color the scatter plot,[filename/rho/cluster]
+    colorscol: The column number of the properties used for the coloring. Starts from 0.
+    peratom: Whether to output per atom pca coordinates (True/False)
+    dimension: The number of principle components to keep
+    pc1: int, default is 0, which principle axis to plot the projection on
+    pc2: int, default is 1, which principle axis to plot the projection on
+    algorithm: the algorithm for density-based clustering options are: ([dbscan], [fdb])
+    projectatomic: build the projection using the (big) atomic descriptor matrix
+    plotatomic: Plot the PCA coordinates of all atomic environments (True/False)
+    adtext: Whether to adjust the text (True/False)
 
     Returns
     -------
 
     """
 
-    if fmat == 'none' and kmat == 'none':
+    use_atomic_desc = (peratom or plotatomic or projectatomic)
+
+    # try to read the xyz file
+    if fxyz != 'none':
+        asapxyz = ASAPXYZ(fxyz)
+        desc, desc_atomic = asapxyz.get_descriptors(fmat, use_atomic_desc)
+        if projectatomic:
+            desc = desc_atomic.copy()
+    else:
+        asapxyz = None
+        print("Did not provide the xyz file. We can only output descriptor matrix.")
+        output = 'matrix'
+
+    if fmat == 'none' and kmat == 'none' and fxyz == 'none':
```

Comment:
I have added a check for an xyz file because presumably we don't want this to throw an error in the case that the user has provided an xyz file.